### PR TITLE
React

### PR DESCRIPTION
**Pull request by eduardavercosa**
- Start the FrontEnd feature with React

### CODE REVIEW

File: backend/duda/urls.py [hunk 1]
```diff
@@ -10,4 +11,9 @@
     path("jsreverse/", django_js_reverse.views.urls_js, name="js_reverse"),
     path("", include("battling.urls")),
     path("api/", include("battling.api.urls")),
+    path(
+        "react/battle/<int:pk>/detail",
```
Comment (rsarai) [inline backend/duda/urls.py]: What do you think of changing this prefix to something more meaningful? And extracting react URLs to another file?

File: backend/duda/urls.py [hunk 2]
```diff
@@ -10,4 +11,9 @@
...
+        TemplateView.as_view(template_name="react/react_template.html"),
+        name="react_template",
```
Comment (rsarai) [inline backend/duda/urls.py]: What about renaming this to remove the `react` prefix?

File: frontend/js/utils/api.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import axios from 'axios';
+
+const baseUrl = window.location.host;
+
+const apiUrls = {
+  battleDetail: (id) => `api/battles/${id}`,
+};
+
+const getFromApi = (urlApi) => {
+  const url = `http://${baseUrl}/${urlApi}`;
```
Comment (rsarai) [inline frontend/js/utils/api.js]: We are using `django-js-reverse` to pass urls from django to react. Instructions on how to set this here: https://github.com/ierror/django-js-reverse#use-the-urls-in-javascript

File: frontend/js/utils/api.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import axios from 'axios';
+
+const baseUrl = window.location.host;
+
+const apiUrls = {
```
Comment (rsarai) [inline frontend/js/utils/api.js]: This is not required since we are using `django-js-reverse`

**Review by rsarai — CHANGES_REQUESTED** [review]

File: backend/battling/v2/urls_v2.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+from django.conf.urls import include, url  # noqa
+from django.urls import path
+from django.views.generic import TemplateView
+
+
+urlpatterns = [
+    path(
+        "battle/<int:pk>/detail",
+        TemplateView.as_view(template_name="react/spa_template.html"),
+        name="spa_template",
```
Comment (rsarai) [inline backend/battling/v2/urls_v2.py]: The name of the url should be something meaningful, is this case: `battle_detail_v2`

File: backend/duda/settings/base.py [hunk 6]
```diff
@@ -35,7 +35,13 @@ def base_dir_join(*args):
 
 POKE_API_URL = "https://pokeapi.co/api/v2/pokemon/"
 
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS=['*']
+CORS_ALLOW_ALL_ORIGINS = True
```
Comment (rsarai) [inline backend/duda/settings/base.py]: Why are you changing this? This is changing the config for all our environments, we usually don't want to allow all hosts for the production server

File: frontend/js/utils/api.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import axios from 'axios';
+
+import Urls from './urls';
+
+const baseUrl = window.location.host;
+
+const urlApi = Urls['battle-detail'](':id');
+
+const getFromApi = () => {
```
Comment (rsarai) [inline frontend/js/utils/api.js]: Shouldn't this receive the `urlApi` parameter?

**Review by rsarai — CHANGES_REQUESTED** [review]
Minor comments

File: frontend/js/utils/api.js (hunk 7 above)
Comment (eduardavercosa) [reply frontend/js/utils/api.js]: It doesn't work if I put the parameter

File: backend/battling/v2/urls_v2.py (hunk 5 above)
Comment (eduardavercosa) [reply backend/battling/v2/urls_v2.py]: I did it in the show battle detail PR

File: backend/duda/settings/base.py (hunk 6 above)
Comment (eduardavercosa) [reply backend/duda/settings/base.py]: I removed it in the show battle detail PR

**Review by rsarai — APPROVED** [review]

**Pull request merged**